crates/matrix-sdk-ui/src/sync_service.rs

```diff
@@ -67,69 +67,40 @@ pub enum State {
     Error,
 }
 
-struct SyncServiceInner {
-    room_list_service: Arc<RoomListService>,
-    encryption_sync_service: Arc<EncryptionSyncService>,
-    state: SharedObservable<State>,
-    encryption_sync_permit: Arc<AsyncMutex<EncryptionSyncPermit>>,
-    /// Supervisor task ensuring proper termination.
-    ///
-    /// This task is waiting for a `TerminationReport` from any of the other two
-    /// tasks, or from a user request via [`Self::stop()`]. It makes sure
-    /// that the two services are properly shut up and just interrupted.
-    ///
-    /// This is set at the same time as the other two tasks.
-    supervisor_task: Option<JoinHandle<()>>,
+/// A supervisor that starts two sync tasks, one for the room list and one for
+/// the end-to-end encryption support.
+struct SyncTaskSupervisor {
+    /// The task that supervises the two sync tasks.
+    task: JoinHandle<()>,
```

Comment:
The comment is confusing at first, because it mentions two tasks but there's only one set in the struct.

Reply:
I attempted to reword this in 7707997dc3a92c2d1d3c390da90837bc833797cb.